queue.go

```diff
@@ -355,52 +357,46 @@ func (queue *redisQueue) consumerConsume(consumer Consumer) {
 
 func (queue *redisQueue) consumerBatchConsume(batchSize int, timeout time.Duration, consumer BatchConsumer) {
 	batch := []Delivery{}
-	timer := time.NewTimer(timeout)
-	stopTimer(timer) // timer not active yet
+	for {
+		// Wait for first delivery
+		delivery, ok := <-queue.deliveryChan
+		if !ok {
+			// debug("batch channel closed") // COMMENTOUT
+			return
+		}
+		batch = append(batch, delivery)
+		// debug(fmt.Sprintf("batch consume added delivery %d", len(batch))) // COMMENTOUT
+		batch, ok = queue.batchTimeout(batchSize, batch, timeout)
+		if !ok {
+			// debug("batch channel closed") // COMMENTOUT
+			return
```

Comment:
I know we currently never close `deliveryChan`, but you seem to be handling that case anyway. I wonder if we shouldn't still process the last batch in that case. So basically always `Consume` the batch from `batchTimeout`, but then only continue with this loop if `ok`. What do you think?

Reply:
I think that is a good catch. I think it's a good idea to have the consumer finish the available batch.